apps/cli/src/ethereum/contract-abi.ts

```diff
@@ -65,4 +71,13 @@ export class ContractAbi {
   hasFunction(name: string): boolean {
     return this.raw.some((t) => t.type === "function" && t.name === name);
   }
+
+  selectorForFn(name: string): Option<Hex> {
+    const found = this.raw.find((t) => t.type === "function" && t.name === name);
+
+    return Option.fromNullable(found)
+      .filter(isAbiFunction)
+      .map((abiFn) => abiFn as AbiFunction)
```

Comment:
Is this necessary with the previous typeguard?

Reply:
I refactored this to make better use of the type predicate and try to make it cleaner.